File: Lab/SEM4/MPP/lab7/moto_java/client/src/main/java/org/zdroba/service/RacerService.java
package org.zdroba.service;

import org.zdroba.entity.RaceEvent;
import org.zdroba.entity.Racer;
import org.zdroba.entity.Team;
import org.zdroba.exceptions.AlreadyExistsException;
import org.zdroba.exceptions.NotFoundException;

import java.util.List;

public interface RacerService {

    List<Racer> find(Team team);

    List<Racer> getAll();

    List<Racer> getAll(int engine);

    void add(String nume, String cnp, Integer engine, Team team) throws AlreadyExistsException, NotFoundException;

    void modify(Long id, Team team) throws NotFoundException;
}

File: Lab/SEM4/MPP/lab6/moto_c/service/RacerService.cs
using moto_c.entity;

namespace moto_c.service;

public interface RacerService
{
    List<Racer> find(Team team);

    List<Racer> getAll();

    List<Racer> getAll(int engine);

    void add(string nume, string cnp, int engine, Team team);
    
    void modify(long id, Team team);
}